definitions/npm/styled-components_v2.x.x/flow_v0.53.x-/styled-components_v2.x.x.js

```diff
@@ -0,0 +1,372 @@
+// @flow
+
+type $npm$styledComponents$Interpolation = (<C: {}>(executionContext: C) => string) | string | number;
+type $npm$styledComponents$NameGenerator = (hash: number) => string;
+
+type $npm$styledComponents$TaggedTemplateLiteral<R> = {| (Array<string>, $npm$styledComponents$Interpolation): R |};
+
+// ---- FUNCTIONAL COMPONENT DEFINITIONS ----
+type $npm$styledComponents$ReactComponentFunctional<Props, DefaultProps = *> = {
+  (Props): React$Node,
+  defaultProps: DefaultProps,
+}
+
+type $npm$styledComponents$ReactComponentFunctionalUndefinedDefaultProps<Props> = { (Props): React$Node }
+
+// ---- CLASS COMPONENT DEFINITIONS ----
+class $npm$styledComponents$ReactComponent<Props, DefaultProps> extends React$Component<Props> {
+  static defaultProps: DefaultProps
+}
+type $npm$styledComponents$ReactComponentClass<Props, DefaultProps = *> = Class<$npm$styledComponents$ReactComponent<Props, DefaultProps>>
+type $npm$styledComponents$ReactComponentClassUndefinedDefaultProps<Props> = Class<React$Component<Props, *>>
+
+// ---- COMPONENT FUNCTIONS INPUT (UNION) & OUTPUT (INTERSECTION) ----
+type $npm$styledComponents$ReactComponentUnion<Props> =
+  | $npm$styledComponents$ReactComponentFunctional<Props>
+  | $npm$styledComponents$ReactComponentFunctionalUndefinedDefaultProps<Props>
```

Comment:
Integration tests (i.e. me copypasting this into a project that uses `styled-components`) are showing that Flow can't pick between `$npm$styledComponents$ReactComponentFunctionalUndefinedDefaultProps<Props>` and `React$StatelessFunctionalComponent<Props>` when `styled` gets passed a functional component.  Have you not run into this error?

Reply:
Can you show the test?

Reply:
```
styled(() => <div />);
```